lib/model/migrations/20240806-01-add-userpreferences.js

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const up = (db) => db.schema.createTable('userpreferences', (prefs) => {
+  prefs.integer('userId').notNull();
+  prefs.jsonb('body');
+
+  prefs.foreign('userId').references('users.actorId');
+  prefs.primary('userId');
+});
+
```

Comment:
We've had some long discussions about what these migrations should look like in the past year, e.g. using knex vs. writing raw sql. More of the recent migrations use raw sql because we have dreams of getting unstuck from Knex (an old db package we happen to be locked into for migrations) and "rewriting the knex migrations to not use knex" was a thought we entertained. I personally don't think that approach will pan out, so I don't really care if the migrations are like this or not. If you happen to have any brilliant ideas on how to get away from knex migrations (while still allowing people to smoothly upgrade their Central setups in the wild), plz share!